### PR TITLE
Update to taskgraph 3.2.0

### PR DESCRIPTION
**Pull request by jcristau**
This updates taskcluster-taskgraph to 3.2.0, along with the decision-mobile docker image used in decision tasks.
Also includes some cleanups in our .taskcluster.yml and ac_taskgraph modules.

### Pull Request checklist

- [ ] **Quality**: This PR builds and passes detekt/ktlint checks (A pre-push hook is recommended)
- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Changelog**: This PR includes [a changelog entry](https://github.com/mozilla-mobile/android-components/blob/main/docs/changelog.md) or does not need one
- [ ] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/main/android/accessibility_guide.md) or does not include any user facing features

### After merge
- **Milestone**: Make sure issues closed by this pull request are added to the [milestone](https://github.com/mozilla-mobile/android-components/milestones) of the version currently in development.
- **Breaking Changes**: If this is a breaking change, please push a draft PR on [Reference Browser](https://github.com/mozilla-mobile/reference-browser) to address the breaking issues.

### GitHub Automation

Used by GitHub Actions.

### CODE REVIEW

File: taskcluster/ac_taskgraph/worker_types.py [hunk 1]
```diff
@@ -5,13 +5,9 @@
 
 from voluptuous import Any, Required, Optional
 
-from taskgraph.util.schema import taskref_or_string, WHITELISTED_SCHEMA_IDENTIFIERS
+from taskgraph.util.schema import taskref_or_string
 from taskgraph.transforms.task import payload_builder
 
-WHITELISTED_SCHEMA_IDENTIFIERS.extend([
-    lambda path: "[u'artifact-map']" in path,   # handed directly in beetmover
```
Comment (JohanLorenzo) [inline taskcluster/ac_taskgraph/worker_types.py:12]: I'm glad we don't use this confusing list of lambdas! 👍

File: .taskcluster.yml [hunk 2]
```diff
@@ -240,7 +216,7 @@ tasks:
                             # Note: This task is built server side without the context or tooling that
                             # exist in tree so we must hard code the hash
                             image:
-                                mozillareleases/taskgraph:decision-mobile-7e11b0cc3966ad9729e08b19551399b3343d3b385eac067b6335d4c34431a899@sha256:b309fa59efd59991ba286a326cb43b724c38e6f3872c52d0f85e96428899c2fc
+                                mozillareleases/taskgraph:decision-mobile-625975b642c148be4c6f1d8ee5cedf7399f5d0dd33d275ff69d5934e3082d4a9@sha256:bfb26700182486e1c6c52701baea6f386fa39e5e25417423c27845933605ad43
```
Comment (JohanLorenzo) [inline .taskcluster.yml:219]: Nit: There's a slightly more up to date image I published on Dockerhub, 2 weeks ago. It notably fixes[1]

```suggestion
                                mozillareleases/taskgraph:decision-mobile-3c208673efd0df8ff6aa174916582a851b5a3e7f40e07f0a446902cca5ed838f@sha256:68fcb0e64e736eb883e5ae61347a45f59eeb78875a09aa56cbf0fe73dfdfdeeb
```

[1] https://github.com/taskcluster/taskgraph/commit/b072703a3165f7ce76c3c713ea889feca83adf3b

**Review by JohanLorenzo — APPROVED** [review]
Thank you very much for removing `github-releases` from the expected graphs now that we ship A-C through shipit! 

This looks good to me modulo a minor nit.

File: .taskcluster.yml (hunk 2 above)
Comment (jcristau) [reply .taskcluster.yml]: Done, thanks.

**Pull request merged**